golang/synthesiserConnector/main.go

```diff
@@ -0,0 +1,397 @@
+					},
+				},
+			},
+		)
+		if err != nil {
+			log.Fatalf("Could not Describe Feed: %v", err)
+		}
+
+		feedValueLabel := feedDescription.GetPayload().Result.GetValues()[0].Label
+
+		feedListener, err := apiContext.InterestAPI.FetchInterests(
+			apiContext.CtxWithMeta,
+			&ioticsApi.FetchInterestRequest{
+				Headers: &apiContext.Headers,
+				Args: &ioticsApi.FetchInterestRequest_Arguments{
+					Interest: &ioticsApi.Interest{
+						FollowerTwinId: &ioticsApi.TwinID{Id: twinThermostatIdentity.Did()},
+						FollowedFeedId: &ioticsApi.FeedID{
+							Id:     feedOfInterest.FeedId.Id,
+							TwinId: feedOfInterest.FeedId.TwinId,
+							HostId: feedOfInterest.FeedId.HostId,
+						},
+					},
+				},
+			},
+		)
+		if err != nil {
+			log.Fatalf("Could not receive Feed messages: %v", err)
+		}
+
+		go getFeedData(feedListener, feedOfInterest.FeedId.TwinId, feedValueLabel, ch)
+	}
+
+	/*** SEARCH FOR TWIN RADIATOR ***/
+	headersSearch = ioticsApi.Headers{
+		ClientAppId:    "Iotics Search",
+		TransactionRef: []string{"Search"},
+		RequestTimeout: timestamppb.New(time.Now().Add(time.Second * 3)),
+	}
+
+	searchResponseClient, searchErr = apiContext.SearchAPI.SynchronousSearch(
+		apiContext.CtxWithMeta,
+		&ioticsApi.SearchRequest{
+			Headers: &headersSearch,
+			Scope:   ioticsApi.Scope_GLOBAL,
+			Payload: &ioticsApi.SearchRequest_Payload{
+				ResponseType: ioticsApi.ResponseType_FULL,
+				Filter: &ioticsApi.SearchRequest_Payload_Filter{
+					Properties: []*ioticsApi.Property{
+						{
+							Key: helpers.PROPERTY_KEY_LABEL,
+							Value: &ioticsApi.Property_LangLiteralValue{
+								LangLiteralValue: &ioticsApi.LangLiteral{
+									Value: "Twin Radiator",
+									Lang:  "en",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	)
+	if searchErr != nil {
+		log.Fatalf("Could not search: %v", searchErr)
+	}
+
+	var twinRadiatorList []*ioticsApi.SearchResponse_TwinDetails
+
+	for {
+		resp, respErr := searchResponseClient.Recv()
+		if respErr != nil {
+			break
+		}
+		twinRadiatorList = append(twinRadiatorList, resp.GetPayload().GetTwins()...)
+	}
+
+	/*** BONUS ***/
+	// _, err = apiContext.TwinAPI.UpsertTwin(
+	// 	apiContext.CtxWithMeta,
+	// 	&ioticsApi.UpsertTwinRequest{
+	// 		Headers: &apiContext.Headers,
+	// 		Payload: &ioticsApi.UpsertTwinRequest_Payload{
+	// 			TwinId: &ioticsApi.TwinID{Id: twinThermostatIdentity.Did()},
+	// 			Properties: []*ioticsApi.Property{
+	// 				{
+	// 					Key: helpers.PROPERTY_KEY_LABEL,
+	// 					Value: &ioticsApi.Property_LangLiteralValue{
+	// 						LangLiteralValue: &ioticsApi.LangLiteral{
+	// 							Value: "Twin Thermostat",
+	// 							Lang:  "en",
+	// 						},
+	// 					},
+	// 				},
+	// 			},
+	// 			Inputs: []*ioticsApi.UpsertInputWithMeta{
+	// 				{
+	// 					Id: "presence",
+	// 					Values: []*ioticsApi.Value{
+	// 						{
+	// 							Label:    "presence_detected",
+	// 							Comment:  "Whether the sensor detected a presence",
```

Comment:
should the input not be something like `turn on thermostat`?
presence detected seems like the condition after which an input is sent to `turn on thermostat`

Reply:
ok, I get that. I think we should be careful about the language used though. 
An input is an action, not a state. 

I think comment should be more like: "Inform thermostat of sensor state"?